### PR TITLE
_Orthogonal.right_inverse cleanup

### PR DESCRIPTION
**Pull request by redwrasse**
Cleans up some of the control flow of the make_orthogonal / is_orthogonal operations in `_Orthogonal.right_inverse`.

### CODE REVIEW

File: torch/nn/utils/parametrizations.py [hunk 1]
```diff
@@ -163,29 +173,16 @@ def right_inverse(self, Q: torch.Tensor) -> torch.Tensor:
...
-            neg_Id = torch.zeros_like(Q_init)
-            neg_Id.diagonal(dim1=-2, dim2=-1).fill_(-1.0)
-            return neg_Id
+
+        self.base = _ensure_tall_square_orthogonal(Q)
+
+        # It is necessary to return the -Id, as we use the diagonal for the
+        # Householder parametrization. Using -Id makes:
+        # householder(torch.zeros(m,n)) == torch.eye(m,n)
+        # Poor man's version of eye_like
```
Comment (vadimkantorov) [inline torch/nn/utils/parametrizations.py:182]: Related on `eye_like`: 
- https://github.com/pytorch/pytorch/issues/9244
- https://github.com/pytorch/pytorch/issues/68703

Comment (redwrasse) [reply torch/nn/utils/parametrizations.py:182]: Thanks @vadimkantorov, I'll look at those issues. Consolidating the APIs inside and outside of the `parameterizations` module would also help make parameterizations more approachable I think.

**Review by lezcano — CHANGES_REQUESTED** [review]
This just moves code around, and it's not clear to me that the newer version is clearer than the older one in any way.

**Comment by redwrasse** [comment]
This MR improves readability and code quality, with minimal changes to the logic. Removes excess if/else nested statements, removes redundant variables, and (related) the scope of inputs to the original helper methods (only expects a tall matrix, hence checking A^H A = Id) is unclear without referring to the context.

Additionally,

- Removes the unused `eps` argument from `_is_orthogonal`.
- Consolidates the `_make_orthogonal` and `_is_orthogonal` operations, since they're used in a specific grouped logic anyways that is specific to the input A being a tall matrix, within the `rightInverse` method.
- Removes the redundant `else` statement since the above `if` statement has a return at the end, in the `rightInverse` method.

**Pull request closed without merging**